common/persistence/sql/sqlTaskStore.go

```diff
@@ -70,7 +68,8 @@ func (m *sqlTaskStore) LeaseTaskList(
 ) (*persistence.LeaseTaskListResponse, error) {
 	var rangeID int64
 	var ackLevel int64
-	shardID := m.shardID(request.DomainID, request.TaskList)
+	shardID := sqlplugin.GetDBShardIDFromDomainIDAndTasklist(request.DomainID, request.TaskList, m.db.GetTotalNumDBShards())
```

Comment:
It breaks our logic in Docstore. the shard id is currently ignore by mysql and postgres, but it's used in our internal docstore db.

Reply:
Just create a new variable dbShardID like in sqlExecutionStore.go and keep the old shardID the same  as before

Reply:
Yeah that will work too but that will be confusing. 
This shardID is calculated by 
`id := farm.Hash32([]byte(domainID+"_"+name)) % uint32(m.nShards)`
which is exactly the same thing that we are trying to get as dbShardID. 

It's not historyShardID.

Reply:
> It breaks our logic in Docstore
I just moved the logic to `dbSharding.go` and the behavior the exactly the same. It shouldn't break anything.

Reply:
Ah, you're right. I just noticed that in our internal config, we set the nShards to be the same as numHistoryShards.